example/tests/zkgroup-test.ts

```diff
@@ -661,134 +663,129 @@ export const testZkGroup = () => {
+    );
+
+    const masterKey = new GroupMasterKey(TEST_ARRAY_32_1);
+    const groupSecretParams = GroupSecretParams.deriveFromMasterKey(masterKey);
+
+    const aliceCiphertext = new ClientZkGroupCipher(
+      groupSecretParams
+    ).encryptServiceId(aliceAci);
+    const groupCiphertexts = [aliceAci, bobAci, eveAci, malloryAci].map(
+      (next) =>
+        new ClientZkGroupCipher(groupSecretParams).encryptServiceId(next)
+    );
+
+    // Server
+    const now = Math.floor(Date.now() / 1000);
+    const startOfDay = now - (now % SECONDS_PER_DAY);
+    const expiration = startOfDay + 2 * SECONDS_PER_DAY;
+    const todaysKey = GroupSendDerivedKeyPair.forExpiration(
+      new Date(1000 * expiration),
+      serverSecretParams
+    );
+    const response = GroupSendEndorsementsResponse.issue(
+      groupCiphertexts,
+      todaysKey
+    );
+
+    // Client
+    const receivedEndorsements = response.receiveWithServiceIds(
+      [aliceAci, bobAci, eveAci, malloryAci],
+      aliceAci,
+      groupSecretParams,
+      serverPublicParams
+    );
+    // Missing local user
+
+    assert(
+      throwsSync(() =>
+        response.receiveWithServiceIds(
+          [bobAci, eveAci, malloryAci],
+          aliceAci,
+          groupSecretParams,
+          serverPublicParams
+        )
+      ),
+      "Missing local user"
+    );
+    // Missing another user
+    assert(
+      throwsSync(() =>
+        response.receiveWithServiceIds(
+          [aliceAci, eveAci, malloryAci],
+          aliceAci,
+          groupSecretParams,
+          serverPublicParams
+        )
+      ),
+      "Missing another user"
+    );
+
+    // Try the other receive too
+    {
+      const receivedEndorsementsAlternate = response.receiveWithCiphertexts(
         groupCiphertexts,
-        todaysKey
-      );
-  
-      // Client
-      const receivedEndorsements = response.receiveWithServiceIds(
-        [aliceAci, bobAci, eveAci, malloryAci],
-        aliceAci,
-        groupSecretParams,
+        aliceCiphertext,
         serverPublicParams
       );
+      assert(
+        deepEql(
+          receivedEndorsements.combinedEndorsement.serialized,
+          receivedEndorsementsAlternate.combinedEndorsement.serialized
+        )
+      );
+
       // Missing local user
       assert(
         throwsSync(() =>
-          response.receiveWithServiceIds(
-            [bobAci, eveAci, malloryAci],
-            aliceAci,
-            groupSecretParams,
+          response.receiveWithCiphertexts(
+            groupCiphertexts.slice(1),
+            aliceCiphertext,
             serverPublicParams
           )
         ),
         "Missing local user"
-      )
+      );
       // Missing another user
+
       assert(
         throwsSync(() =>
-          response.receiveWithServiceIds(
-            [aliceAci, eveAci, malloryAci],
-            aliceAci,
-            groupSecretParams,
+          response.receiveWithCiphertexts(
+            groupCiphertexts.slice(0, -1),
```

Comment:
you can't change api by only changing the ios files. currently the android tests are failing. this should be handled inside the native function. revert all changes to the test file